### PR TITLE
Fix command with complete but no nargs (#724)

### LINKED ISSUE #724: VSCode starts with error message: "E1208: -complete used without -nargs"
When I start vscode, I get the following message:

```bash
VSCode-Neovim:

line   34:
E1208: -complete used without -nargs
line   10:
E1208: -complete used without -nargs
```

The problem occurs even if the `init.vim` is empty.

## Versions

VSCode

```txt
Version: 1.59.0
Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
Date: 2021-08-04T23:13:20.182Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Linux x64 5.11.0-25-generic
```

neovim

```txt
NVIM v0.6.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -g -O2 -fdebug-prefix-map=/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9=. -fstack-protector-strong -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9/build/config -I/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9/src -I/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9/.deps/usr/include -I/usr/include -I/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9/build/src/nvim/auto -I/build/neovim-0bOHCD/neovim-0.5.0+ubuntu2+git202108111755-7d2233fad-d569569c9/build/include
Compiled by buildd@lgw01-amd64-059

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### PR DESCRIPTION
**Pull request by gen4438**
This PR fixes #724.

### CODE REVIEW

File: vim/vscode-file-commands.vim [hunk 1]
```diff
@@ -31,7 +31,7 @@ command! -bang -nargs=? Ex call <SID>editOrNew(<q-args>, <q-bang>)
 command! -bang Enew call <SID>editOrNew('__vscode_new__', <q-bang>)
 command! -bang Find call VSCodeNotify('workbench.action.quickOpen')
 
-command! -complete=file -bang Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
+command! -bang Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
```
Comment (rtmor) [inline vim/vscode-file-commands.vim:34]: This appears to work, however, are there any advantages to removing the `-complete=file` parameter over simply including the `nargs=?` parameter? Both appear to fix the issue.

```suggestion
command! -complete=file -bang -nargs=? Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
```

File: vim/vscode-tab-commands.vim [hunk 2]
```diff
@@ -7,7 +7,7 @@ function! s:switchEditor(...) abort
 endfunction
 
 command! -complete=file -nargs=? Tabedit if empty(<q-args>) | call VSCodeNotify('workbench.action.quickOpen') | else | call VSCodeExtensionNotify('open-file', expand(<q-args>), 0) | endif
-command! -complete=file Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
+command! Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
```
Comment (rtmor) [inline vim/vscode-tab-commands.vim:10]:
```suggestion
command! -complete=file -nargs=? Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
```

**Review by rtmor — COMMENTED** [review]
These changes appear to work, however, is there an advantage to removing the `-complete=file` parameter entirely, as opposed to appending `nargs=?` to each command?

**Comment by gen4438** [comment]
Thank you for your review! Your suggestion looks better than my changes, so I've incorporated them.

(Actually, I modified it as you suggested at first, but since the default value of `: command-nargs` of nvim is` -nargs = 0` and does not accept arguments, I committed it so that the behavior does not change.)

**Comment by mcaple** [comment]
Would be really nice if someone could approve this

**Comment by GerardKetuma** [comment]
> Would be really nice if someone could approve this

Yep, also waiting on this to be approved. My :w does not work and also getting same E1208 output messages.

**Comment by Kilill** [comment]
Tested this, seems to work fine, would someone please commit it ?, so that i don't have to patch it every time reinstall vscode :-)

**Comment by mcaple** [comment]
Sorry @asvetliakov but any chance of reviewing this?

**Comment by jsonMartin** [comment]
Also getting this error, looking forward to a merge of this 👍

**Comment by francois-pasquier** [comment]
is there some update on this?

**Review by Kilill — APPROVED** [review]
Aplied this and has been working without a hitch

**Comment by Kilill** [comment]
For the impatient , I applied the patch manually by directly editing the files works quite ok ( you find the files in your <Home Directory>/.vscode/extenstions/asvetliakov.vscode-neovim-0.0.82/vim

**Review by asvetliakov — APPROVED** [review]

**Comment by asvetliakov** [comment]
Thank you for the fix!

**Comment by willopez** [comment]
@asvetliakov can we get a new version publish for this update? that would be great, thanks!

**Comment by asvetliakov** [comment]
Published v0.0.83

**Pull request merged**